src/main.js

```diff
@@ -52,6 +52,33 @@ export async function main(platform) {
         });
         await vm.load();
         platform.createAndMountRootView(vm);
+
+        // needed for mobile Safari which does *not* resize the viewport when the keyboard
+        // shows or hides (see https://bugs.webkit.org/show_bug.cgi?id=141832)
+        if (window.visualViewport) {
+            window.visualViewport.addEventListener('resize', () => {
+                const sessionView = document.getElementsByClassName('SessionView')[0];
+                if (!sessionView) {
+                    return;
+                }
+
+                const timeline = document.getElementsByClassName('Timeline')[0];
+                let scrollTopBefore, heightBefore, heightAfter;
+
+                if (timeline) {
+                    scrollTopBefore = timeline.scrollTop;
+                    heightBefore = timeline.getBoundingClientRect().height;
+                }
+
+                sessionView.style.height = window.visualViewport.height.toString() + 'px';
+                sessionView.style.top = window.visualViewport.offsetTop.toString() + 'px';
+
+                if (timeline) {
+                    heightAfter = timeline.getBoundingClientRect().height;
+                    timeline.scrollTop = scrollTopBefore + heightBefore - heightAfter;
```

Comment:
Hmm, wonder if we could easily make this not specific to the timeline ... like do it for all scrollable elements or something :thinking: